lib/ecto/changeset.ex

```diff
@@ -2941,7 +2941,7 @@ defimpl Inspect, for: Ecto.Changeset do
     end
 
     redacted_fields = case data do
-      %type{} -> type.__schema__(:redact_fields)
+      %type{} -> if function_exported?(type, :__schema__, 2), do: type.__schema__(:redact_fields), else: []
```

Comment:
I tried to use pattern matching but couldn't get it to work. There is probably a much better way to solve this.

Reply:
You can check if it also has a `__meta__: ...` field instead of function_exported?. WDYT?

Reply:
```suggestion
      %type{__meta__: _} -> type.__schema__(:redact_fields)
```